pkg/resources/management.cattle.io/v3/project/quota_validate.go

```diff
@@ -0,0 +1,47 @@
+package project
+
+import (
+	mgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/data/convert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	quotav1 "k8s.io/apiserver/pkg/quota/v1"
+)
+
+// quotaFits checks whether the quota in the second argument is sufficient for the requested quota in the first argument.
+// If it is not sufficient, a list of the resources that exceed the allotment is returned.
+// The ResourceList to be checked can be compiled by passing a ResourceQuotaLimit to convertLimitToResourceList,
+// or a slice of ResourceQuotaLimit to convertLimitsToResourceList, before calling this function on the result.
+func quotaFits(resourceList corev1.ResourceList, projectLimit *mgmtv3.ResourceQuotaLimit) (bool, corev1.ResourceList, error) {
```

Comment:
Design nit: for simplicity, we may want to have both parameters be of the same resourceList type (as the comment implies). The conversion of the second parameter to a resourceList could happen outside the function. As a result, we would be able so also not return an error from it and simplify it further.

Reply:
Done. This made more sense when this was sometimes working with slices.